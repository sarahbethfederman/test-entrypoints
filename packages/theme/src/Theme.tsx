import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
