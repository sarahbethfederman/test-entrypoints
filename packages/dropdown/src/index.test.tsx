import * as React from 'react';
import { mount } from 'enzyme';
import Theme from '@lendi-ui/theme';
import Dropdown from './index';
import { DropdownWrapper, Select, IconDown } from './index.style';

let Wrapper;

describe('Dropdown', () => {
  beforeEach(() => {
    const items = [
      {
        value: '1',
        label: '5 years',
      },
      {
        value: '2',
        label: '10 years',
      },
      {
        value: '3',
        label: '15 years',
      },
      {
        value: '4',
        label: '20 years',
      },
      {
        value: '5',
        label: '25 years',
      },
    ];

    Wrapper = mount(
      <Theme>
        <Dropdown size="md" items={items} />
      </Theme>
    );
  });

  it('render whole component', () => {
    expect(Wrapper.find('Dropdown').length).toEqual(1);
    expect(Wrapper.find('Dropdown')).toMatchSnapshot();
  });

  it('render DropdownWrapper', () => {
    expect(Wrapper.find(DropdownWrapper).length).toEqual(1);
    expect(Wrapper.find(DropdownWrapper)).toMatchSnapshot();
  });

  it('render Dropdown component and has size property "md"', () => {
    expect(Wrapper.find(Select).length).toEqual(1);
    expect(Wrapper.find(Select).prop('selectSize')).toEqual('md');
    expect(Wrapper.find(Select)).toMatchSnapshot();
  });

  it('render SelectedItem component', () => {
    expect(Wrapper.find('option').length).toEqual(5);
    expect(Wrapper.find('option')).toMatchSnapshot();
  });

  it('render BoxArrow component', () => {
    expect(Wrapper.find(IconDown).length).toEqual(1);
    expect(Wrapper.find(IconDown)).toMatchSnapshot();
  });
});

describe('Dropdown onChange', () => {
  const onChange = jest.fn();

  const wrapper = mount(
    <Theme>
      <Dropdown size="md" items={[{ value: '1', label: '1' }, { value: '2', label: '2' }]} onChange={onChange} />
    </Theme>
  );

  it('should call onChange prop', () => {
    const select = wrapper.find(Select);
    select.simulate('change', { target: { value: '2' } });
    expect(onChange.mock.calls).toEqual([['2']]);
  });
});
