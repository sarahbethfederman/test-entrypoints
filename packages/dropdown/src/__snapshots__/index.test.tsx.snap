// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown it should render disable when isDisabled = true  1`] = `null`;

exports[`Dropdown it should render error when isError = true  1`] = `null`;

exports[`Dropdown it should render whole component 1`] = `
.c3 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
}

.c3 > svg {
  fill: #00c0a5;
  width: inherit;
  height: inherit;
}

.c0 {
  display: inline-block;
  position: relative;
  width: 100%;
}

.c0:hover {
  box-shadow: 0 12px 17px 2px rgba(0,0,0,0.1), 0 5px 22px 4px rgba(0,0,0,0.12), 0 7px 8px 0 rgba(0,0,0,0.14);
}

.c1 {
  border-radius: 4px 4px 0 0;
  font-family: 'Open Sans',sans-serif;
  -webkit-appearance: none;
  cursor: pointer;
  padding: calc(0.4 * var(--lendi-ui-size)) calc(2.5 * var(--lendi-ui-size)) calc(0.4 * var(--lendi-ui-size)) calc(0.66 * var(--lendi-ui-size));
  font-size: var(--lendi-ui-size);
  width: 100%;
  border: none;
}

.c2 {
  position: absolute;
  right: 6px;
  pointer-events: none;
  top: calc(0.45 * var(--lendi-ui-size));
  width: calc(1.25 * var(--lendi-ui-size));
  height: calc(1.25 * var(--lendi-ui-size));
}

<Dropdown
  items={
    Array [
      Object {
        "label": "5 years",
        "value": "1",
      },
      Object {
        "label": "10 years",
        "value": "2",
      },
      Object {
        "label": "15 years",
        "value": "3",
      },
      Object {
        "label": "20 years",
        "value": "4",
      },
      Object {
        "label": "25 years",
        "value": "5",
      },
    ]
  }
  size="md"
>
  <styled.div
    isDisabled={false}
    isFullWidth={true}
  >
    <div
      className="c0"
    >
      <styled.select
        disabled={false}
        isError={false}
        onChange={[Function]}
        selectSize="md"
      >
        <select
          className="c1"
          disabled={false}
          onChange={[Function]}
        >
          <option
            key="1"
            value="1"
          >
            5 years
          </option>
          <option
            key="2"
            value="2"
          >
            10 years
          </option>
          <option
            key="3"
            value="3"
          >
            15 years
          </option>
          <option
            key="4"
            value="4"
          >
            20 years
          </option>
          <option
            key="5"
            value="5"
          >
            25 years
          </option>
        </select>
      </styled.select>
      <Styled(ChevronDown)
        color="primary.500"
        size="md"
      >
        <ChevronDown
          className="c2"
          color="primary.500"
          size="md"
        >
          <styled.span
            className="c2"
            color="primary.500"
            size="md"
          >
            <span
              className="c2 c3"
              color="primary.500"
              size="md"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41z"
                />
              </svg>
            </span>
          </styled.span>
        </ChevronDown>
      </Styled(ChevronDown)>
    </div>
  </styled.div>
</Dropdown>
`;
